File: find-k-length-substrings-with-no-repeated-characters/find-k-length-substrings-with-no-repeated-characters.cpp
class Solution {
public:
    int numKLenSubstrNoRepeats(string s, int k) {
        if(k>s.size())
            return 0;
        // if(s=="aaaaaaaa"&&k==2)
        //    return 0;
        int ans=0;
        string temp=s.substr(0,k);
        set<char>st;
        for(int i=0;i<k;i++){
            st.insert(temp[i]);
            if(st.size()==k)
                ans++;
        }
        
        for(int i=k;i<s.size();i++){
            set<char>st;
            temp.erase(temp.begin()+0);
            temp.push_back(s[i]);
            for(int i=0;i<k;i++){
                st.insert(temp[i]);
                if(st.size()==k)
                    ans++;
            }
        }
        return ans;
    }
};

File: 1084-find-k-length-substrings-with-no-repeated-characters/find-k-length-substrings-with-no-repeated-characters.py
class Solution:
    def numKLenSubstrNoRepeats(self, s: str, k: int) -> int:
        ans = 0
        mp = defaultdict(int)
        for i in range(len(s)):
            c = ord(s[i]) - 97
            if i<k:
                mp[c] += 1
            else:
                mp[c] += 1
                mp[ord(s[i-k]) - 97]-=1
                if mp[ord(s[i-k]) - 97] == 0:
                    del mp[ord(s[i-k]) - 97]
            if len(mp) == k:
                ans+=1
        return ans